test/core/tsi/alts/handshaker/alts_target_name_override_test.cc

```diff
@@ -0,0 +1,284 @@
+
+void* tag(int i) { return (void*)static_cast<intptr_t>(i); }
+
+class FakeHandshakeServer {
+ public:
+  FakeHandshakeServer(bool check_num_concurrent_rpcs,
+                      const std::string& expected_target_name) {
+    int port = grpc_pick_unused_port_or_die();
+    address_ = grpc_core::JoinHostPort("localhost", port);
+    service_ = grpc::gcp::CreateFakeHandshakerService(
+        0 /* expected max concurrent rpcs unset */, expected_target_name);
+    grpc::ServerBuilder builder;
+    builder.AddListeningPort(address_.c_str(),
+                             grpc::InsecureServerCredentials());
+    builder.RegisterService(service_.get());
+    server_ = builder.BuildAndStart();
+    gpr_log(GPR_INFO, "Fake handshaker server listening on %s",
+            address_.c_str());
+  }
+
+  ~FakeHandshakeServer() {
+    server_->Shutdown(grpc_timeout_milliseconds_to_deadline(0));
+  }
+
+  const char* address() { return address_.c_str(); }
+
+ private:
+  std::string address_;
+  std::unique_ptr<grpc::Service> service_;
+  std::unique_ptr<grpc::Server> server_;
+};
+
+// Perform a simple RPC and capture the value of the authority header
+// metadata sent to the server, as a string.
+std::string PerformCallAndGetAuthorityHeader(grpc_channel* channel,
+                                             grpc_server* server,
+                                             grpc_completion_queue* cq) {
+  grpc_call* c;
+  grpc_call* s;
+  cq_verifier* cqv = cq_verifier_create(cq);
+  grpc_op ops[6];
+  grpc_op* op;
+  grpc_metadata_array initial_metadata_recv;
+  grpc_metadata_array trailing_metadata_recv;
+  grpc_metadata_array request_metadata_recv;
+  grpc_call_details call_details;
+  grpc_status_code status;
+  grpc_call_error error;
+  grpc_slice details;
+  gpr_timespec deadline = grpc_timeout_seconds_to_deadline(5);
+  grpc_slice request_payload_slice = grpc_slice_from_copied_string("request");
+  grpc_byte_buffer* request_payload =
+      grpc_raw_byte_buffer_create(&request_payload_slice, 1);
+  grpc_slice response_payload_slice = grpc_slice_from_copied_string("response");
+  grpc_byte_buffer* response_payload =
+      grpc_raw_byte_buffer_create(&response_payload_slice, 1);
+  grpc_byte_buffer* request_payload_recv = nullptr;
+  grpc_byte_buffer* response_payload_recv = nullptr;
+  // Start a call
+  c = grpc_channel_create_call(channel, nullptr, GRPC_PROPAGATE_DEFAULTS, cq,
+                               grpc_slice_from_static_string("/foo"), nullptr,
+                               deadline, nullptr);
+  GPR_ASSERT(c);
+  grpc_metadata_array_init(&initial_metadata_recv);
+  grpc_metadata_array_init(&trailing_metadata_recv);
+  grpc_metadata_array_init(&request_metadata_recv);
+  grpc_call_details_init(&call_details);
+  memset(ops, 0, sizeof(ops));
+  op = ops;
+  op->op = GRPC_OP_SEND_INITIAL_METADATA;
+  op->data.send_initial_metadata.count = 0;
+  op->flags = 0;
+  op->reserved = nullptr;
+  op++;
+  op->op = GRPC_OP_SEND_MESSAGE;
```

Comment:
None of the send_message or recv_message ops are actually needed in this test.

Reply:
done, removed send and receive message ops